File: Server/ChatServer/src/main/java/com/szlazakm/chatserver/dtos/KeyBundleDTO.java
package com.szlazakm.chatserver.dtos;

import lombok.Builder;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;

@Value
@Builder
@Jacksonized
public class KeyBundleDTO {

    String identityKey;

    int signedPreKeyId;
    String signedPreKey;
    String signature;

    Integer onetimePreKeyId;
    String onetimePreKey;
}

File: client/app/src/main/java/com/szlazakm/safechat/webclient/dtos/KeyBundleDTO.kt
package com.szlazakm.safechat.webclient.dtos

import com.fasterxml.jackson.annotation.JsonProperty

data class KeyBundleDTO (
    @JsonProperty("identityKey")
    val identityKey: String,
    @JsonProperty("signedPreKeyId")
    val signedPreKeyId: Int,
    @JsonProperty("signedPreKey")
    val signedPreKey: String,
    @JsonProperty("signature")
    val signature: String,
    @JsonProperty("onetimePreKeyId")
    val onetimePreKeyId: Int?,
    @JsonProperty("onetimePreKey")
    val onetimePreKey: String?
)
